plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -656,13 +656,6 @@ private void resolveTemplateVariablesIfAny(KubernetesList resources) throws Mojo
             }
             File kubernetesYaml = new File(this.targetDir, "kubernetes.yml");
             resolveTemplateVariables(parameters, kubernetesYaml);
-            File kubernetesResourceDir = new File(this.targetDir, "kubernetes");
```

Comment:
Could you please elaborate on how this fixes #1732

Reply:
oh, so we were resolving placeholders in template files also.

Reply:
Those lines were replacing template placeholders in kubernetes resource files with concrete values specified in template.yaml file. The placeholders need to be resolved only in kubernetes.yml in order to be able to do fabric8:deploy or fabric8:apply, but they don't need to be resolved in individual kubernetes resource files as they will be needed in HelmMojo execution.
The HelmMojo is using those same kubernetes resources to create chart, but since the placeholders were already replaced with concrete values from template.yaml, it can not replace them with corresponding `.Values.xxx` placeholders.

Reply:
Easiest is to build spring-boot-with-yaml sample on this branch with fabric:helm goal and inspect generated helm chart